### ISSUE TITLE
do we need `max_num_children`?

### ISSUE THREAD

**Issue by pstjohn** [NREL/graph-env#26]
it would be ideal if we could support an action space without explicitly padding to max_num_children.
We could evaluate the next action softmax with something like tenorflow's 
[`tf.RaggedTensor.from_row_lengths`](https://www.tensorflow.org/api_docs/python/tf/RaggedTensor#from_row_lengths) and `tf.nn.softmax`, for instance.

I found this PR, how much of what we've done overlaps with their built-in feature?
https://github.com/ray-project/ray/pull/8393
https://docs.ray.io/en/latest/rllib/rllib-models.html#variable-length-complex-observation-spaces

https://discuss.ray.io/t/rllib-variable-length-observation-spaces-without-padding/726/6
https://groups.google.com/g/ray-dev/c/_cBxgfHTYZw

**Comment by pstjohn** [comment]
Some other links:
https://github.com/ray-project/ray/blob/master/rllib/utils/spaces/repeated.py

Looks like they link to these scripts that do the batching / unbatching:
https://github.com/ray-project/ray/blob/master/rllib/models/repeated_values.py

Although it looks like they still use a max_len parameter, so not sure if that reflects well on our ability to get around it

**Comment by pstjohn** [comment]
Largely solved by moving to the Repeated action space and by masking inputs to the forward model
